ci-operator/step-registry/cucushift/installer/rehearse/aws/cases/clusters/provision/cucushift-installer-rehearse-aws-cases-clusters-provision-commands.sh

```
@@ -290,10 +296,20 @@ if [ "${TEST_OBJECT}" == "LocalZones" ] || [ "${TEST_OBJECT}" == "WavelengthZone
+    echo "EdgeZone column is empty or absent, discovering edge zones from AWS API ..."
+    EDGE_ZONES_JSON=/tmp/edge_zones.json
+    jq -r --arg t "$ZONE_TYPE" '[.AvailabilityZones[] | select(.ZoneType==$t)]' "${ZONES_JSON}" > "${EDGE_ZONES_JSON}"
 
-  EDGE_ZONE_COUNT=$(jq -r '. | length' ${EDGE_ZONES_JSON})
-  EDGE_ZONE_NAMES=$(jq -r '.[].ZoneName' ${EDGE_ZONES_JSON})
+    EDGE_ZONE_COUNT=$(jq -r '. | length' ${EDGE_ZONES_JSON})
+    EDGE_ZONE_NAMES=$(jq -r '.[].ZoneName' ${EDGE_ZONES_JSON})
+  else
+    echo "Using edge zones from spreadsheet: ${EDGE_ZONE_NAMES_FROM_SHEET}"
+    EDGE_ZONE_NAMES=$(echo "${EDGE_ZONE_NAMES_FROM_SHEET}" | tr ',' '\n' | sed 's/^[[:space:]]*//' | sed '/^$/d' | tr '\n' ' ')
+    EDGE_ZONE_COUNT=$(echo ${EDGE_ZONE_NAMES} | wc -w)
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)